polaris-react/.storybook/GridPanel/GridPanel.tsx

```diff
@@ -0,0 +1,85 @@
+import React, {useMemo} from 'react';
+import {useGlobals, useGlobalTypes, useParameter} from '@storybook/api';
+import {
+  AddonPanel,
+  ArgsTable,
+  ArgTypes,
+  PresetColor,
+} from '@storybook/components';
+
+export type GlobalControlsParam = {
+  presetColors?: PresetColor[];
+};
+
+type PanelProps = {
+  active: boolean;
+};
+
+const filterUncontrolledTypes = (globalTypes: ArgTypes) =>
+  Object.entries(globalTypes).reduce((acc, [key, arg]) => {
+    if (arg.control !== undefined) acc[key] = arg;
+    return acc;
+  }, {} as ArgTypes);
```

Comment:
```suggestion
const filterUncontrolledTypes = (globalTypes: ArgTypes) =>
  Object.fromEntries(
    Object.entries(globalTypes)
      .map(([key, arg]) => arg.control ? [key, arg] : null)
      .filter(Boolean),
  ) as ArgTypes;
```

Light suggestion: Untested but I've been encouraging folks to use `fromEntries()` instead of constructing objects with `reduce()` where possible.